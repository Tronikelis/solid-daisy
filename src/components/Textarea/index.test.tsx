/* eslint-disable solid/reactivity */
import { render } from "@solidjs/testing-library";
import userEvent from "@testing-library/user-event";
import { createSignal } from "solid-js";
import { describe, expect, it } from "vitest";

import { Textarea } from ".";

describe("<Textarea />", () => {
    it("renders correctly", () => {
        const { container } = render(() => <Textarea />);
        expect(container).toMatchSnapshot();
    });

    it("controlled input works", async () => {
        const [value, setValue] = createSignal("");

        const screen = render(() => (
            <Textarea
                data-testid="textarea"
                value={value()}
                onInput={e => setValue(e.target.value)}
            />
        ));

        await userEvent.type(screen.getByTestId("textarea"), "foobar");

        expect(value()).toBe("foobar");
        expect(screen.getByTestId("textarea")).toHaveValue("foobar");

        setValue("bar");
        expect(screen.getByTestId("textarea")).toHaveValue("bar");
    });
});
