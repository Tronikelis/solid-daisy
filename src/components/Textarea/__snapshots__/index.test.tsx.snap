// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Textarea /> > renders correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-4"
  >
    <textarea
      class="textarea text-inherit"
    />
  </div>
</div>
`;
